### ISSUE TITLE
Azure Speech SDK performance issues while processing audio in “mu law” format.

### ISSUE THREAD

**Issue by jnbh48** [Azure-Samples/cognitive-services-speech-sdk#1681]
**Describe the bug**
We observe performance issues with Azure Speech SDK during processing audio in “mu law” format in our ASP.NET Core application running in the Kubernetes environment.
We use Azure Speech SDK Client - Microsoft.CognitiveServices.Speech , version 1.23.0, and gstreamer component: 
libgstreamer1.0-0 gstreamer1.0-plugins-base gstreamer1.0-plugins-good installed on pod.
Pod resources:
          limits:
            cpu: 4500m
            memory: 2Gi
          requests:
            cpu: "4"
            memory: 1500Mi

Problems arise with more than 18 concurrent audio sessions to  Azure Speech resource. We see High CPU usage (close to 100  %), memory usage about  40%. 
Is this the expected performance? Can we change something in the SDK configuration?
A similar a 100 % CPU usage issue posted on the https://stackoverflow.com/questions/5530074/cpu-usage-and-memory-of-gstreamer-pipleline
It is  related to data sink gstreamer configuration, and setup flag sync=true.  How does the Azure SDK configure the sync flag? 

**To Reproduce**
1. Run more than 18 concurrent audio in “mu law” format sessions using Azure Speech SDK to  Azure Speech resource.

**Expected behavior**
Run more than 18 audio in “mu law” format sessions with success.

**Version of the Cognitive Services Speech SDK**
version 1.23.0

**Platform, Operating System, and Programming Language**
 - OS: [ Linux ] - in the Kubernetes environment
 - Hardware - amd64 / x86_64
 - Programming language: C#

**Additional context**
I attached the SDK client logs for 20 concurrent audio sessions when the issue arose.
[azureSdkLogFile20streams.zip](https://github.com/Azure-Samples/cognitive-services-speech-sdk/files/9707659/azureSdkLogFile20streams.zip)

**Comment by jhakulin** [comment]
@jnbh48 Thanks for the report, in the coming v1.24.0 Speech SDK release (ETA mid Oct 2022), a-law & mu-law formats can be sent directly to the speech service without Gstreamer involvement. This should help with the performance as no decoding to PCM is needed for alaw/mulaw formats.

**Comment by dargilco** [comment]
@jnbh48 are these live audio streams that you are recognizing?

**Comment by jnbh48** [comment]
@dargilco yes, we are recognizing live (real-time) audio streams.

**Comment by jnbh48** [comment]
@jhakulin What is reason for the CPU load issue, decoding mu-law formats to PCM using Gstreamer, is it visible in sdk logs? Can we change something in the SDK configuration?

**Comment by jhakulin** [comment]
@jnbh48 I am not sure if the CPU load issue you have is due to Gstreamer, do you see the issue if you make similar test using PCM streams instead of mu-law?

If the CPU load issue is primarily caused by Gstreamer, in 1.24.0 Speech SDK release which has ETA mid Oct 2022, you can configure Speech SDK to stream mu-law directly to the speech service. This should help to reduce the CPU load because there will not be Gstreamer doing the decoding to PCM.

```
auto format = AudioStreamFormat::GetWaveFormat(8000, 16, 1, AudioStreamWaveFormat::MULAW);
// Creates a push stream
auto pushStream = AudioInputStream::CreatePushStream(format);
// Creates a speech recognizer from stream input;
auto audioInput = AudioConfig::FromStreamInput(pushStream);
auto recognizer = SpeechRecognizer::FromConfig(config, audioInput);
```

**Comment by WGK864** [comment]
Hey, I am a developer who works alongside @jnbh48. We are seeing very worrying results for both mulaw and pcm format.
One of the issues as described above is that one instance uses a lot of cpu when mulaw is used due to what we think is gstreamer. The PCM tests are clearly using a lot less of the cpu resources which we allocate.
Another issue that we are seeing is that no matter how much we spread the traffic over many instances, the max concurrent call count is around 25 concurrent transcriptions. We are testing this theory by running the same amount of transcriptions over two speech services to verify whether a single instance is somehow limiting us. Will get back to you with results once that is done.

**Comment by jhakulin** [comment]
@WGK864 @jnbh48 1.24 speech sdk is now available which supports direct streaming of mulaw (no need for gstreamer). Would you give us update on this issue?

**Comment by WGK864** [comment]
Hey, version 1.24 seemed to have fixed the CPU issue. The other issue (scaling) is tracked in an Azure Enterprise Support ticket.

**Comment by jhakulin** [comment]
thanks for the feedback, closing the issue.
